`nomad monitor` always panics

### Nomad version
Nomad v0.10.2 (0d2d6e3dc5a171c21f8f31fa117c8a765eb4fc02)

### Operating system and Environment details
```
$ cat /etc/os-release 
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux
```
### Issue
`nomad monitor` always fails with the following output, no matter what log-level and node-id I use:

```
$ nomad monitor -log-level=INFO -node-id=3d651214-cc4c-fe06-fe5c-9834ef11148c
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/hashicorp/nomad/command.(*MonitorCommand).Run(0xc00002cf20, 0xc0000b8060, 0x2, 0x2, 0x0)
	/opt/gopath/src/github.com/hashicorp/nomad/command/agent_monitor.go:114 +0x1192
github.com/hashicorp/nomad/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc0004c68c0, 0xc0004c68c0, 0xc0002e0500, 0x38)
	/opt/gopath/src/github.com/hashicorp/nomad/vendor/github.com/mitchellh/cli/cli.go:255 +0x1f1
main.RunCustom(0xc0000b8050, 0x3, 0x3, 0xc00008e058)
	/opt/gopath/src/github.com/hashicorp/nomad/main.go:138 +0x4a1
main.Run(...)
	/opt/gopath/src/github.com/hashicorp/nomad/main.go:83
main.main()
	/opt/gopath/src/github.com/hashicorp/nomad/main.go:79 +0x63
```

Some nodes in the cluster are still on 0.9, but I don't try to get logs from them.

### Reproduction steps

Call `nomad monitor` with any arguments.

### Job file (if appropriate)

### Nomad Client logs (if appropriate)
### Nomad Server logs (if appropriate)

No new log lines appear after that command.

--- a/command/agent_monitor.go
+++ b/command/agent_monitor.go
@@ -106,6 +106,11 @@ func (c *MonitorCommand) Run(args []string) int {
 			return 1
 		}
 
+		if len(nodes) == 0 {
+			c.Ui.Error(fmt.Sprintf("No node(s) with prefix or id %q found", nodeID))
+			return 1
+		}
+
 		if len(nodes) > 1 {
 			out := formatNodeStubList(nodes, false)
 			c.Ui.Output(fmt.Sprintf("Prefix matched multiple nodes\n\n%s", out))

--- a/command/agent_monitor_test.go
+++ b/command/agent_monitor_test.go
@@ -14,7 +14,7 @@ func TestMonitorCommand_Implements(t *testing.T) {
 
 func TestMonitorCommand_Fails(t *testing.T) {
 	t.Parallel()
-	srv, _, _ := testServer(t, false, nil)
+	srv, _, url := testServer(t, false, nil)
 	defer srv.Shutdown()
 
 	ui := new(cli.MockUi)
@@ -33,4 +33,13 @@ func TestMonitorCommand_Fails(t *testing.T) {
 	if code := cmd.Run([]string{"-address=nope"}); code != 1 {
 		t.Fatalf("exepected exit code 1, got: %d", code)
 	}
+
+	// Fails on nonexistent node
+	if code := cmd.Run([]string{"-address=" + url, "-node-id=12345678-abcd-efab-cdef-123456789abc"}); code != 1 {
+		t.Fatalf("expected exit 1, got: %d", code)
+	}
+	if out := ui.ErrorWriter.String(); !strings.Contains(out, "No node(s) with prefix") {
+		t.Fatalf("expected not found error, got: %s", out)
+	}
+	ui.ErrorWriter.Reset()
 }

